spark/src/main/kotlin/com/adevinta/spark/res/AnnotatedStringResource.kt

```diff
@@ -0,0 +1,200 @@
+        .substringAfter("<p dir=\"ltr\">").substringBeforeLast("</p>")
+}
+
+/**
+ * The framework `getText()` method doesn't support formatting arguments, so we need to do it ourselves.
+ */
+internal fun Resources.getText(@StringRes id: Int, vararg args: Any): CharSequence {
+    val escapedArgs = args.map {
+        if (it is Spanned) it.toHtmlWithoutParagraphs() else it
+    }.toTypedArray()
+    val spannedString = SpannedString(getText(id))
+    val htmlResource = spannedString.toHtmlWithoutParagraphs()
+    val formattedHtml = String.format(htmlResource, *escapedArgs)
+    return HtmlCompat.fromHtml(formattedHtml, HtmlCompat.FROM_HTML_MODE_LEGACY)
+}
+
+private fun CharSequence.asAnnotatedString(
+    density: Density,
+    colors: SparkColors,
+    typography: SparkTypography,
+): AnnotatedString {
+    if (this !is Spanned) return AnnotatedString(this.toString())
+    return buildAnnotatedString {
+        append(this@asAnnotatedString.toString())
+        getSpans(0, length, Any::class.java).forEach {
+            val start = getSpanStart(it)
+            val end = getSpanEnd(it)
+            buildWithSpan(it, start, end, density, colors, typography)
+        }
+    }
+}
+
+private fun AnnotatedString.Builder.buildWithSpan(
+    it: Any,
+    start: Int,
+    end: Int,
+    density: Density,
+    colors: SparkColors,
+    typography: SparkTypography,
+) {
+    val span: SpanStyle = when (it) {
+        is StyleSpan -> buildWithStyleSpan(it)
+        is TypefaceSpan -> buildWithTypefaceSpan(it)
+        is BulletSpan -> {
+            Log.d("StringResources", "BulletSpan not supported yet")
+            return
+        }
+
+        is AbsoluteSizeSpan -> with(density) {
+            SpanStyle(fontSize = if (it.dip) it.size.dp.toSp() else it.size.toSp())
+        }
+
+        is RelativeSizeSpan -> SpanStyle(fontSize = it.sizeChange.em)
+        is StrikethroughSpan -> SpanStyle(textDecoration = TextDecoration.LineThrough)
+        is UnderlineSpan -> SpanStyle(textDecoration = TextDecoration.Underline)
+        is SuperscriptSpan -> SpanStyle(baselineShift = BaselineShift.Superscript)
+        is SubscriptSpan -> SpanStyle(baselineShift = BaselineShift.Subscript)
+        is ForegroundColorSpan -> SpanStyle(color = Color(it.foregroundColor))
+        is Annotation -> SparkStringAnnotations.toSpanStyle(it.key, it.value, colors, typography)
+
+        else -> return
+    }
+    addStyle(span, start, end)
+}
+
+private fun buildWithStyleSpan(it: StyleSpan): SpanStyle {
+    return when (it.style) {
+        Typeface.NORMAL -> FontWeight.Normal to FontStyle.Normal
+        Typeface.BOLD -> FontWeight.Bold to FontStyle.Normal
+        Typeface.ITALIC -> FontWeight.Normal to FontStyle.Italic
+        Typeface.BOLD_ITALIC -> FontWeight.Bold to FontStyle.Italic
+        else -> null
+    }?.let { (fontWeight, fontStyle) ->
+        SpanStyle(fontWeight = fontWeight, fontStyle = fontStyle)
+    } ?: SpanStyle()
+}
```

Comment:
To keep the same naming, should we also update like this? (and remove the empty span)

```suggestion
private fun StyleSpan.to SpanStyle(it: StyleSpan): SpanStyle? {
    return when (it.style) {
        Typeface.NORMAL -> FontWeight.Normal to FontStyle.Normal
        Typeface.BOLD -> FontWeight.Bold to FontStyle.Normal
        Typeface.ITALIC -> FontWeight.Normal to FontStyle.Italic
        Typeface.BOLD_ITALIC -> FontWeight.Bold to FontStyle.Italic
        else -> null
    }?.let { (fontWeight, fontStyle) ->
        SpanStyle(fontWeight = fontWeight, fontStyle = fontStyle)
    }
}
```